src/index.ts

```diff
@@ -180,7 +180,7 @@ export class UnleashClient extends TinyEmitter {
         const staticContext = {environment: this.context.environment, appName: this.context.appName };
         this.context = {...staticContext, ...context};
         if (this.timerRef) {
-            this.fetchToggles();
+            await this.fetchToggles();
```

Comment:
That is the thing @ivarconr 
I've just tested this against 2 applications and if I remove the await from there, this function will execute instantly and will report as resolved instantly as well ( because otherwise updateContext doesn't know when fetchToggles has finished the request and the storage update )
If we leave that await there, this function will get resolved whenever this.fetchToggles is resolved, just when you would expect ( when the flags are loaded with the new context which was updated )

Reply:
I think if we were to return the toggles fetched, then the await would be implicit and we could just `return this.fetchToggles()` promise. But because updateContext doesn't return the flags, then we need that await there as it is not implicit...

Reply:
makes sense!